integrations/lib/testing/integration/authhelper.go

```diff
@@ -0,0 +1,197 @@
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package integration
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/constants"
+	"github.com/gravitational/teleport/api/identityfile"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/keys"
+	libauth "github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+	"testing"
+	"time"
+)
+
+// OSSAuthHelper implements the AuthHelper interface.
+// It starts an OSS Auth server and exposes the functions required for plugin
+// integration tests to build teleport clients for each user/plugin/bot.
+type OSSAuthHelper struct {
+	server *libauth.TestTLSServer
+	dir    string
+}
+
+// StartServer implements the AuthHelper interface.
```

Comment:
Can we super clear in the godoc what (if anything) needs to be manually closed and what is automatically cleaned up?